cognite/gqlpygen/main.py

```diff
@@ -1,60 +1,165 @@
+        click.echo("Searching for a schema...")
+        for schema_name, instance in inspect.getmembers(module):
+            if isinstance(instance, Schema):
+                click.echo(f"Found schema {schema_name!r}")
+                break
+        else:
+            click.echo("Failed to find schema, exiting..")
+            exit(1)
     else:
-        click.echo("Failed to find schema, exiting..")
-        exit(1)
+        schema_name = to_schema_name(prefix)
+        click.echo(f"Got schema name {schema_name}")
+        instance = getattr(module, schema_name)
 
-    output = Path.cwd() / "schema.graphql"
-    output.write_text(instance.as_str())
-    click.echo(f"Wrote file {output}")
+    graphql_schema.write_text(instance.as_str())
+    click.echo(f"Wrote file {graphql_schema}")
+
+
+@app.command("signin", help="Upload a GQL schema to CDF DM data model.")
+def signin(
+    cdf_cluster: str = typer.Option(_cdf_cluster or ..., help="CDF cluster name."),
+    tenant_id: str = typer.Option(_tenant_id or ..., help="AD tenant ID."),
+    client_id: str = typer.Option(_client_id or ..., help="AD client ID."),
+    client_secret: str = typer.Option(
+        _client_secret_hidden or None, prompt=True, hide_input=True, help="AD client secret."
+    ),
+    project: str = typer.Option(_project or ..., help="Name of CDF project."),
+):
+    if client_secret == _client_secret_hidden:
+        client_secret = _client_secret
+    if client_secret == _client_secret_none:
+        auth_option = "--device-code"
+        auth_option_hidden = None
+    else:
+        auth_option = f"--client-secret='{client_secret}'"
+        auth_option_hidden = f"--client-secret='{client_secret[:3]}{'*' * (len(client_secret) - 3)}'"
+
+    command = [
+        "cdf",
+        "signin",
+        f"'{project}'",
+        f"--cluster='{cdf_cluster}'",
+        f"--tenant='{tenant_id}'",
+        f"--client-id='{client_id}'",
+        auth_option,
+    ]
+    msg = f"Executing:\n{' '.join(command)}"
+    if auth_option_hidden:
+        msg = msg.replace(auth_option, auth_option_hidden)
+    typer.echo(f"Executing:\n{msg}")
+    subprocess.call(command)
+
+
+@app.command("upload", help="Upload a GQL schema to CDF DM data model.")
+def upload(
+    graphql_schema: Path = typer.Argument(_graphql_schema or ..., help="GraphQL schema to upload."),
+    space: str = typer.Option(_space or ..., help="Space ID in CDF Domain Modeling"),
+    data_model: str = typer.Option(_datamodel or ..., help="ID of Data Model in CDF Domain Modeling"),
+    schema_version: int = typer.Option(_schema_version or ..., help="Version of the schema to app or update."),
+):
+    command = [
+        "cdf",
+        "data-models",
+        "publish",
+        f"--file='{graphql_schema}'",
+        f"--space='{space}'",
+        f"--external-id='{data_model}'",
+        f"--version='{schema_version}'",
+    ]
+    typer.echo(f"Executing:\n{' '.join(command)}")
+    subprocess.call(command)
```

Comment:
Catch error of the `cdf` CLI is not installed and give a user friendly message, with installation instructions.